types/query/filtered_pagination.go

```diff
@@ -85,6 +85,16 @@ func FilteredPaginate(
 	end := offset + limit
 
 	for ; iterator.Valid(); iterator.Next() {
+		if numHits == end+1 {
```

Comment:
Ah I got this wrong right here, because it looks at `iterator.Key()`. (Which is a different value in this loop)

This is a bit annoying to make consistent with the old behavior. Will adjust the de-duplicated loop to get this consistent

Reply:
Basically the old behavior would ideally worked with `numHits = end` right here.

But it actually made nextKey skip to the next hit, which is kind of annoying to get to work.

Reply:
The current PR has this problem fixed.

In a later release when redesigning the Filtered paginate, I would recommend changing this behavior. (As it will both be more correct from a performance expectation perspective, and code quality cleanup. We could delete all the extra for loops everywhere.)